nncf/experimental/torch/fx/transformations.py

```diff
@@ -528,6 +542,128 @@ def fuse_conv_bn(model: torch.fx.GraphModule) -> None:
+
+
+def _remove_constant_qdq_transformation(model: torch.fx.GraphModule) -> None:
+    def match_filters(match, original_graph, graph):
+        for node in match.nodes_map:
+            if node.name == "weight" and match.nodes_map[node].op == "get_attr":
+                return True
+        return False
+
+    pattern, replacement = _get_pattern_replacement_per_channel()
+    matches_per_channel = torch.fx.subgraph_rewriter.replace_pattern_with_filters(
+        model, pattern, replacement, [match_filters]
+    )
+    _set_meta_for_matches(model, matches_per_channel)
+    pattern, replacement = _get_pattern_replacement_per_tensor()
+    matches_per_tensor = torch.fx.subgraph_rewriter.replace_pattern_with_filters(
+        model, pattern, replacement, [match_filters]
+    )
+    _set_meta_for_matches(model.graph, matches_per_tensor)
+
+
+def _get_node_inputs(node: torch.fx.Node, model: torch.fx.GraphModule):
+    input_tup = []
+    for params in node.args:
+        if isinstance(params, torch.fx.Node):
+            if params.op == "get_attr":
+                value = get_tensor_constant_from_node(params, model)
+                input_tup.append(value)
+            else:
+                return None
+        else:
+            input_tup.append(params)
+    return input_tup
+
+
+def _compress_qdq_constant_transformation(model: torch.fx.GraphModule) -> None:
+    for node in model.graph.nodes:
+        if node.target in [
+            torch.ops.quantized_decomposed.quantize_per_tensor.default,
+            torch.ops.quantized_decomposed.quantize_per_channel.default,
+        ]:
+            port_id = 0
+            input_tup = _get_node_inputs(node, model)
+            if input_tup:
+                if node.target == torch.ops.quantized_decomposed.quantize_per_channel.default:
+                    _reshape_scale(model, node)
+                result = node.target(*tuple(input_tup))
+                constant_update_fn(model, node, result, port_id, updated_node_name="compressed_weight_updated_constant")
+
+
+def _reshape_scale(model, node: torch.fx.Node):
+    weight_node = node.all_input_nodes[0]
+    scale_node = node.all_input_nodes[1]
+    scale_value = get_tensor_constant_from_node(scale_node, model)
+    weight_value = get_tensor_constant_from_node(weight_node, model)
+    axis = node.args[3]
+    new_shape = [1] * weight_value.dim()
+    new_shape[axis] = scale_value.shape[0]
+    scale_value = scale_value.reshape(new_shape)
+    constant_update_fn(model, node, scale_value, 1, updated_node_name=scale_node.name + "_updated_constant")
+
+
+def fq_weights_transformation(model: torch.fx.GraphModule) -> None:
+    for node in model.graph.nodes:
+        if node.target in [
+            torch.ops.quantized_decomposed.dequantize_per_tensor.default,
+            torch.ops.quantized_decomposed.dequantize_per_channel.default,
+        ]:
+            port_id = 0
+            input_tup = _get_node_inputs(node.all_input_nodes[0], model)
```

Comment:
I would suggest to check that `node.all_input_nodes[0]` is            `torch.ops.quantized_decomposed.quantize_per_tensor.default` or      `torch.ops.quantized_decomposed.quantize_per_channel.default`

Reply:
Done!